### DISCUSSION TITLE
Shared chat improvements.

### DISCUSSION THREAD

**Discussion by treejadey** [Chatterino/chatterino2#5756] [Ideas] [answered]
Seeing somebody use a shared chat between 5 people, made me realise that it becomes really hard to separate the users, because at a glance you can only see if users are in the same chat as you are or not, and you have to **manually hover** over each person to find out where they are from. It of course, works if you're sharing the chat with a single user, but if more, it becomes a bigger problem. 

In my opinion, the solution is to have small, badge-like profile pictures of the chat the user is chatting from. I was personally thinking of implementing this but I would need some help, as I have not contributed code to Chatterino before and I am unfamiliar to the conventions of the project.

---
Out of additional improvements I was thinking that it would be nice if filters had information on what channel the message comes from, not sure if other people would need it though. 

If you have any other potential improvements, you should drop them below. ![smile emoji](https://static-cdn.jtvnw.net/emoticons/v2/1/default/dark/1.0)

**Comment by gempir** [comment]
Maybe we could somehow setup a pre installed Filter for shared chat? 

Filters are great but mostly for advanced users. Chatterino can easily filter shared chat messages but it's somewhat complicated to setup. Maybe a default Filter that filters all messages that are shared could be good? Should probably be available in the UI in Set Filters directly. By default I mean just available as an option not turned on by default. 

And then it's also in the Filters settings. 
This also has the benefit of introducing users to filters in an easy way.

**Reply by Nerixyz** [reply]
I think we should make a setting to disable shared chat completely. The 7TV extension has an option to hide shared chat as well. It could work similar to how "Hide deleted messages" works.

**Comment by jupjohn** [comment]
Support for our internal shared chat badge as a target for a badge highlight could be useful; I'm thinking of a case where you want to change the background colour of external chat messages - although it doesn't solve the problem of 2+ shared chats.

**Comment by 8thony** [comment]
I see more and more shared chat sessions.
The current implementation is a good start, but some points I came up with:
- Show subscription messages from all participating channels.
- Show the sub badge from the channel the message is coming from.
- Mod, VIP,... badges #6653
- Show Hostchannel badge #6244
- Add an easy way to toggle displayed shared chat messages.
	- something like the sword for modview maybe? 
	- predefined filter? (probably needs #3061 first).

**Reply by iProdigy** [reply]
> Show subscription messages from all participating channels.

This was an intentional decision as broadcasters were confused as to who the true recipient of the sub was, but we can explore using the channel icon to make it more clear

**Comment by Nerixyz** [comment]
### Show Participant Names in Split Header

Currently, you have to hover the badges in the chat to see which streamers are part of the shared chat session. We could try to show the participants of a shared chat session in the split header. For example: `streamer (participant1, participant2)`.

We will probably need to poll [Get Shared Chat Session](https://dev.twitch.tv/docs/api/reference/#get-shared-chat-session) for this instead of using EventSub, because of the current subscription limits.

I'm not sure if we can do this in all Stream Together sessions (I don't think we can get the info from Twitch).

**Reply by pajlada** [reply]
Made into an issue https://github.com/Chatterino/chatterino2/issues/6934

**Comment by pajlada** [comment, marked as answer]
As most of the issues from this discussion have been either fixed or had an issue made for them I'll mark this discussion as fixed.
If more shared chat feedback comes up, feel free to create a new discussion.
